### PR TITLE
Use all Mithril lifecycle stubs

### PR DESCRIPTION
**Pull request by askvortsov1**
See https://github.com/flarum/core/issues/2446

### CODE REVIEW

File: js/src/forum/components/Search.js [hunk 1]
```diff
@@ -200,7 +202,8 @@ export default class Search extends Component {
     window.addEventListener('resize', this.updateMaxHeightHandler);
   }
 
-  onremove() {
+  onremove(vnode) {
+    super.onremove(vnode);
     window.removeEventListener('resize', this.updateMaxHeightHandler);
```
Comment (davwheat) [inline js/src/forum/components/Search.js:207]: Just to match the rest of the changes

```suggestion
    super.onremove(vnode);

    window.removeEventListener('resize', this.updateMaxHeightHandler);
```

File: js/src/forum/components/PostStreamScrubber.js [hunk 2]
```diff
@@ -142,7 +144,8 @@ export default class PostStreamScrubber extends Component {
     this.stream.loadPromise.then(() => this.updateScrubberValues({ animate: false, forceHeightChange: true }));
   }
 
-  onremove() {
+  onremove(vnode) {
+    super.onremove(vnode);
     this.scrollListener.stop();
```
Comment (davwheat) [inline js/src/forum/components/PostStreamScrubber.js:149]:
```suggestion
    super.onremove(vnode);

    this.scrollListener.stop();
```

File: js/src/forum/components/Composer.js [hunk 3]
```diff
@@ -56,7 +56,8 @@ export default class Composer extends Component {
     );
   }
 
-  onupdate() {
+  onupdate(vnode) {
+    super.onupdate(vnode);
```
Comment (clarkwinkelmann) [inline js/src/forum/components/Composer.js:60]: Do we also want an additional empty line below the `super` call like all the other instances for consistent formatting? Same thing on line 100 of this file.

**Review by clarkwinkelmann — APPROVED** [review]
Not tested locally, but looks good to me :+1: 

Just one comment below about consistent formatting.

File: js/src/forum/components/Composer.js [hunk 4]
```diff
@@ -95,7 +97,8 @@ export default class Composer extends Component {
       .on('mouseup', (this.handlers.onmouseup = this.onmouseup.bind(this)));
   }
 
-  onremove() {
+  onremove(vnode) {
+    super.onremove(vnode);
     $(window).off('resize', this.handlers.onresize);
```
Comment (davwheat) [inline js/src/forum/components/Composer.js:102]:
```suggestion
    super.onremove(vnode);

    $(window).off('resize', this.handlers.onresize);
```

**Pull request merged**